analyzers/src/SonarAnalyzer.Common/SymbolicExecution/Roslyn/RuleChecks/EmptyCollectionsShouldNotBeEnumeratedBase.cs

```diff
@@ -125,9 +125,9 @@ protected override ProgramState PreProcessSimple(SymbolicContext context)
         {
             return context.State.SetOperationConstraint(operation, CollectionConstraint.Empty);
         }
-        else if (operation.AsInvocation() is { Instance: not null } invocation)
+        else if (operation.AsInvocation() is { } invocation)
```

Comment:
The argument-based logic is hooked in on a wrong place. You should check for argument operation instead. Now two different functionalities are mixed together.